students/mkdir01/lesson09/mailroomOO.py

```diff
@@ -0,0 +1,269 @@
+#!/usr/bin/env python3
+import sys
+import ast
+from datetime import date
+
+
+###
+class Individual_Donor:
+    # _donations = {'Date1': [Donation1], 'Date2': [Donation2, Donation3]}
+
+    def __init__(self, donor_name, donation_amt, donation_date=str(date.today())):
+        self._name = donor_name
+        self._donations = {}
+        self.add_dollars(donation_amt, donation_date)
+
+    @property
+    def name(self):
+        return self._name
+
+    @property
+    def donations(self):
+        return self._donations
+
+    @property
+    def length(self):
+        return len(self.donations)
+
+    def __getitem__(self, key):
+        if key not in self.donations:
+            raise KeyError(key)
+        else:
+            return(self.donations[key])
+
+    def __setitem__(self, key):
+        pass  # use add_dollars instead
+
+    def add_dollars(self, donation_amt, donation_date=str(date.today())):
+        if donation_date in self.donations:
+            self.donations[donation_date].append(float(donation_amt))
+        else:
+            self.donations[donation_date] = [float(donation_amt)]
+
+    def report_total_donations(self):
+        total_amt = 0.00
+        for day in self.donations:
+            for amt in self.donations[day]:
+                total_amt += amt
+        return round(total_amt, 2)
```

Comment:
This could be done a bit more concisely with sum and a list comp. Something like:
```
sum([v for value in self.donations.values() for v in value])
```
But if your donations data structure requires you to use a nested for loop every time you need to make one of these calculations, you might consider reworking it for more efficient access